# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/ash/src/extensions/ext/tooling_info.rs
# Commit: ef5dc3e04e5f
# Full Hash: ef5dc3e04e5f271eea0636ab3a495e95cc912f1d
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2021-09-04 09:40:24
# Regressor Bug: 1726626
# File Overlap Count: 3
# Description:
#   Bug 1726626 - Move gfx/wgpu into a 3rd party dependency r=jgilbert,bholley
#   
#   This update makes wgpu a vendored dependency instead of having it in gfx/wgpu.
#   
#   ## Notes
# ==============================================================================

diff --git a/third_party/rust/ash/src/extensions/ext/tooling_info.rs b/third_party/rust/ash/src/extensions/ext/tooling_info.rs
--- a/third_party/rust/ash/src/extensions/ext/tooling_info.rs
+++ b/third_party/rust/ash/src/extensions/ext/tooling_info.rs
@@ -1,38 +1,37 @@
-#![allow(dead_code)]
 use crate::prelude::*;
-use crate::version::{EntryV1_0, InstanceV1_0};
 use crate::vk;
+use crate::{EntryCustom, Instance};
 use std::ffi::CStr;
 use std::mem;
 use std::ptr;
 
 #[derive(Clone)]
 pub struct ToolingInfo {
     handle: vk::Instance,
     tooling_info_fn: vk::ExtToolingInfoFn,
 }
 
 impl ToolingInfo {
-    pub fn new<E: EntryV1_0, I: InstanceV1_0>(entry: &E, instance: &I) -> ToolingInfo {
+    pub fn new<L>(entry: &EntryCustom<L>, instance: &Instance) -> Self {
         let tooling_info_fn = vk::ExtToolingInfoFn::load(|name| unsafe {
             mem::transmute(entry.get_instance_proc_addr(instance.handle(), name.as_ptr()))
         });
-        ToolingInfo {
+        Self {
             handle: instance.handle(),
             tooling_info_fn,
         }
     }
 
     pub fn name() -> &'static CStr {
         vk::ExtToolingInfoFn::name()
     }
 
-    #[doc = "https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceToolPropertiesEXT.html"]
+    #[doc = "<https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkGetPhysicalDeviceToolPropertiesEXT.html>"]
     pub unsafe fn get_physical_device_tool_properties(
         &self,
         physical_device: vk::PhysicalDevice,
     ) -> VkResult<Vec<vk::PhysicalDeviceToolPropertiesEXT>> {
         let mut count = 0;
         self.tooling_info_fn
             .get_physical_device_tool_properties_ext(physical_device, &mut count, ptr::null_mut())
             .result()?;